metrics/density/footprint_data.sh

```diff
@@ -0,0 +1,359 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A script to gather memory 'footprint' information as we launch more
+# and more containers
+# It allows configuration of a number of things:
+# - which container workload we run
+# - which container runtime we run with
+# - when do we terminate the test (cutoff points)
+#
+# There are a number of things we may wish to add to this script later:
+# - allow configuration of the amount of memory passed to each container
+# - sanity check that the correct number of runtime components (qemu, shim etc.)
+#  are running at all times
+# - clean up better after ourselves (that will be fixed in the generic lib files...)
+# - some post-processing scripts to generate stats and graphs
+#
+# The script gathers information about both user and kernel space consumption
+# Output is into a .csv file, named using some of the config component names
+# (such as results-cc-runtime-busybox.csv)
+
+# Pull in some common, useful, items
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Note that all vars that can be set from outside the script (that is,
+# passed in the ENV), use the ':-' setting to allow being over-ridden
+
+###
+# Define what we will be running (app under test)
+#  Default is we run busybox, as a 'small' workload
+PAYLOAD="${PAYLOAD:-busybox}"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+PAYLOAD_RUNTIME_ARGS="${PAYLOAD_RUNTIME_ARGS:-}"
+
+#  mysql is a more medium sized workload
+#PAYLOAD="${PAYLOAD:-mysql}"
```

Comment:
Commented out lines. If these are generally useful, how about creating a `data/` directory with a `mysql.cfg`, `elasticsearch.cfg`, *etc*` and have a CLI option that will simply `source` one of those data files if specified.

Reply:
I'm not sold on the data dir and breaking the examples out into separate files.
I can make the commented out examples into more obvious examples - and now we have the same data in the doc file, it feels better to make the information parallel/reflect that here - I'll change the format to more comment style